adr/0016-compliance-in-pepr.md

```diff
@@ -0,0 +1,45 @@
+# 14. Compliance in Pepr 
+
+Date: 2025-02-13
+
+
+## Status
+
+Pending
+
+
+## Context
+
+Pepr needs to generate reports for compliance purposes, demonstrating adherence to required security and compliance standards through specific bindings. To achieve this, it’s essential to map compliance controls to Pepr’s bindings. During our discovery process, we determined that Pepr’s role in compliance should focus on reporting rather than enforcement. This is because compliance controls are already enforced through the bindings themselves, and Pepr’s primary function is to report on the compliance status of resources.
+
+## Decision
+
+Extend the fluent API with a new keyword function that accepts a strongly typed control, allowing it to be associated with a given binding. This enables users to map compliance controls to Pepr’s bindings, facilitating the generation of compliance reports based on control status.
+
+
+### Consequences ###
+
+- Extend the Capability class with a new function, such as `CompliesWith()`, that accepts a strongly typed control to associate it with a given binding.
+- Store compliance data in `PeprStore` (or introduce a new store for this purpose).
```

Comment:
Second thought I think we should just use a new `PeprStore` resource like we do with the schedule that is out of scope for the Store actions that are exposed to the Module author instead